Cabal/doc/developing-packages.rst

```diff
@@ -1735,12 +1741,20 @@ system-dependent values for these fields.
     an executable in the package will override any of the hard-coded packages with the same name,
     and a hard-coded package will override any executable on the ``PATH``.
 
-    In the first two cases, the list entry is desguared into a :pkg-field:`build-tool-depends` entry.
+    In the first two cases, the list entry is desugared into a :pkg-field:`build-tool-depends` entry.
     In the first case, the entry is desugared into a :pkg-field:`build-tool-depends` entry by prefixing with ``$pkg:``.
     In the second case, it is desugared by looking up the package and executable name in a hard-coded table.
     In either case, the optional version bound is passed through unchanged.
     Refer to the documentation for :pkg-field:`build-tool-depends` to understand the desugared field's meaning, along with restrictions on version bounds.
 
+    If backwards-compatibility is important, you may opt to use
```

Comment:
The idea is that, with old Cabal, there was no way to actually request the executable be installed, so it was always a case of "user has to manually install executable before you try installing this." So putting the dep in `build-tool-depends` (where it will be ignored) is no worse than the status quo. (And they can put it in `build-depends` if they want to do that horrible hack which only works with `cabal install`)